lib/web_ui/lib/src/engine/view_embedder/dom_manager.dart

```diff
@@ -204,6 +205,33 @@ class DomManager {
       sceneHost.append(sceneElement);
     }
   }
+
+  /// Injects a platform view with [platformViewId] into [platformViewsHost].
+  ///
+  /// If the platform view is already injected, this method does *nothing*.
+  ///
+  /// The `platformViewsHost` can only be different if `platformViewId` is moving
+  /// from one [FlutterView] to another. In that case, the browser will move the
+  /// slot contents from the old `platformViewsHost` to the new one, but that
+  /// will cause the platformView to reset its state (an iframe will re-render,
+  /// text selections will be lost, video playback interrupted, etc...)
+  ///
+  /// Try not to move platform views across views!
+  void injectPlatformView(int platformViewId) {
+    // For now, we don't need anything fancier. If needed, this can be converted
+    // to a PlatformViewStrategy class for each web-renderer backend?
+    try {
+      final DomElement pv = PlatformViewManager.instance.getSlottedContent(platformViewId);
+      // If pv is already a descendant of platformViewsHost -> noop
+      if (pv.parent == platformViewsHost) {
+        return;
+      }
+      platformViewsHost.append(pv);
+    } catch (e) {
+      // Do not fail, many tests expect this to not crash!
+      domWindow.console.warn(e);
+    }
```

Comment:
Not super thrilled about this, but it fixes everything... opinions @mdebbar/@yjbanov?

Reply:
What exactly is throwing here? The only thing I can think of is the assertion in `getSlottedContent`, but I would expect that to work if the tests are setting everything up correctly.

Reply:
It is exactly the `getSlottedContent` assertion (or an unexpected null value if we don't assert). There's a bunch of tests in the framework that are not setting everything up correctly indeed.